eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -193,7 +193,8 @@ function Set-ChangeLogContent {
 
   try
   {
-    $ChangeLogEntries = $ChangeLogEntries.Values | Sort-Object -Descending -Property ReleaseStatus, ReleaseVersion
+    $ChangeLogEntries = $ChangeLogEntries.Values | Sort-Object -Descending -Property ReleaseStatus, `
+      @{e = {[AzureEngSemanticVersion]::new($_.ReleaseVersion)}}
```

Comment:
Can you please test this with a case where the version isn't valid? Will ReleaseVersion be empty or contain the invalid version? I want to be sure that this doesn't end up causing an exception and if there is a failure it just returns null. For that reason it might make sense to switch to the Parse method instead.

Also we might have similar issue with ReleaseStatus as I assume it is a string and so we might want to cast it as a date for sorting purposes as well.

Reply:
Casting it as a date wouldn't work. There's "(Unreleased"). Besides, we use ISO8601 sortable dates, which means they sort lexigraphically, and YYYY (an actual number, that is) sorts before "Unreleased", so reversing that (which we want for dates anyway) works out. This is correct as-is.

Reply:
You are right. The constructor always creates the `AzureEngSemanticVersion` object even when the version is invalid.
`ParseVersionString` returns null when in many cases of invalid version though not all. I think we should stay with the constructor though for two reason. One is that it allows us to automatically apply the correct convention for python as well as other languages. The other is that (Get-ChangeLogEntries)[https://github.com/Azure/azure-sdk-tools/blob/490a400d64315e089c822426221ec6c962a1282e/eng/common/scripts/ChangeLog-Operations.ps1#L27] already ignores invalid version using the `$RELEASE_TITLE_REGEX`

Reply:
OK if we are sure that the version can never be invalid here and thus cause the ctor to throw then I think we are ok here. I also agree with the date sorting as it should work out given the way we are formatting.